com.vrcfury.vrcfury/Editor/VF/Service/OnDemandSyncService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using VF.Builder;
+using VF.Feature.Base;
+using VF.Injector;
+using VF.Utils.Controller;
+using VRC.SDK3.Avatars.Components;
+
+namespace VF.Service {
+    [VFService]
+    public class OnDemandSyncService {
+        [VFAutowired] private readonly AvatarManager manager;
+        [VFAutowired] private readonly MathService math;
+        [VFAutowired] private readonly DirectBlendTreeService directTree;
+        [VFAutowired] private readonly SmoothingService smoothingService;
+        
+        private readonly List<VFAFloat> onDemandSyncParameters = new List<VFAFloat>();
+        
+        public void SetParameterOnDemandSync(VFAFloat parameter) {
+            onDemandSyncParameters.Add(parameter);
+        }
+
+        [FeatureBuilderAction(FeatureOrder.OnDemandSync)]
+        public void Apply() {
+            if (onDemandSyncParameters.Count == 0) return;
+            
+            var fx = manager.GetController(VRCAvatarDescriptor.AnimLayerType.FX);
+            var layers = fx.GetRaw().GetLayers();
+            foreach (var param in onDemandSyncParameters) {
+                manager.GetParams().GetParam(param.Name()).networkSynced = false;
+            }
+            var pointer = fx.NewInt("SyncPointer", synced: true);
+            var localPointer = fx.NewInt("LocalPointer"); // Use a local only pointer to avoid race condition
+            var data = fx.NewFloat("SyncData", synced: true);
+            var sending = fx.NewFloat("IsSending");
+
+            var txLayer = fx.NewLayer("Send");
+            var txIdle = txLayer.NewState("Idle");
+            txIdle.Drives(sending, 0, true);
+            var txAdder = txLayer.NewState("Adder");
+            var txResetter = txLayer.NewState("Reset");
+            txResetter.Drives(localPointer, 0, local: true)
+                .TransitionsToExit()
+                .When(fx.Always());
+            txAdder.TransitionsTo(txResetter)
+                .When(localPointer.IsGreaterThan(onDemandSyncParameters.Count - 1)); // always > 0
+            txAdder.DrivesDelta(localPointer, 1)
+                .TransitionsToExit()
+                .When(fx.Always());
+            
+            // instant sync on change detection
+            for (int i = 0; i < onDemandSyncParameters.Count; i++) {
+                var src = onDemandSyncParameters[i];
+                var lastSrc = fx.NewFloat($"last{src.Name()}");
+                directTree.Add(math.MakeCopier(src, lastSrc)); 
+                var srcDiff = math.Subtract(src, lastSrc);
+                var maintain = math.MakeMaintainer(srcDiff); // Maintain while in send animation
```

Comment:
This won't work right. AAPs should really only be targeted by one animation at a time, which in this case is the Subtract. A variable can't be the output of both a Subtract and a Maintainer

Reply:
Perhaps you want something like
```
var sending = fx.NewFloat("sending{i}");
math.SetValueWithConditions($"diffPending{i}",
  (0, sending.IsGreaterThan(0.5)
  (1, srcDiff.IsGreaterThan(0).Or(srcDiff.IsLessThan(0))
)
```
SetValueWithConditions will maintain if none of the conditions are true.

Reply:
I wasn't able to find a solution without adding another state for each sending parameter. To keep it simple, I set it to 1 when diff != 0 and accumulate that.